### PR TITLE
BUG: fix sharey overwrite on area plots

### LINKED ISSUE #37942: BUG: Area plot overwrites matplotlib's default share y axes behaviour
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
import matplotlib.pyplot as plt

df0 = pd.DataFrame([{'a': 10, 'b': 12}, {'a': 13, 'b': 9}])
df1 = pd.DataFrame([{'a': 100, 'b': 120}, {'a': 130, 'b': 90}])

fig, (axes_0, axes_1) = plt.subplots(1, 2, figsize=(12, 6), sharey='row')
df0.plot(ax=axes_0, kind='area')
df1.plot(ax=axes_1, kind='area')
```

#### Problem description

The area plot overwrites matplotlib's share y axes behaviour if `sharey` parameter is not passed to `.plot` and as a result the second plot get's cut off.
Screenshot 2020-11-18 at 18 55 02

#### Expected Output

same plot made directly with matplotlib, expected behaviour: 
```python
fig, (axes_0, axes_1) = plt.subplots(1, 2, figsize=(12, 6), sharey='row')
axes_0.stackplot(df0.index, df0['a'], df0['b'])
axes_1.stackplot(df1.index, df1['a'], df1['b'])
```
Screenshot 2020-11-18 at 18 56 54

### PR DESCRIPTION
**Pull request by lrusnac**
- [x] closes #37942
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a test which replicates the original behavior and resolves with the patch.

**Comment by lrusnac** [comment]
Hey @jreback, I was looking into that but could not locate any area plots tests, could you suggest a suitable location such a test? is it worth starting a similar file like https://github.com/pandas-dev/pandas/blob/master/pandas/tests/plotting/test_hist_method.py for area_method? if so perhaps that should be bigger task at testing the area plots in general?

**Comment by charlesdong1991** [comment]
could put the tests to `pandas/tests/plotting/frame/test_frame.py` (and `pandas/tests/plotting/test_series.py` if you want to have it tested for `Series`)

**Comment by lrusnac** [comment]
@charlesdong1991 thanks for the pointers, should have looked closer there indeed 😅 
@jreback some tests added now

**Review by jreback — APPROVED** [review]
lgtm @charlesdong1991

**Review by charlesdong1991 — CHANGES_REQUESTED** [review]
Thanks for the PR!

I just have one minor comment on whatsnew, plus a general question: what if users explicitly specify `sharey=False` in `df.plot`? Will `sharey` previously defined by matplotlib be overridden by pandas (I know such a setting is very silly)?

something like below using example from your test:
```python
df = DataFrame(np.random.rand(4, 2), columns=["x", "y"])
fig, (ax1, ax2) = plt.subplots(1, 2, sharey=True)

df.plot(ax=ax1, kind="area", sharey=False)
df.plot(ax=ax2, kind="area", sharey=False)
```

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -664,6 +664,7 @@ Plotting
 - Bug in :meth:`Series.plot` and :meth:`DataFrame.plot` was throwing :exc:`ValueError` with a :class:`Series` or :class:`DataFrame`
   indexed by a :class:`TimedeltaIndex` with a fixed frequency when x-axis lower limit was greater than upper limit (:issue:`37454`)
 - Bug in :meth:`DataFrameGroupBy.boxplot` when ``subplots=False``, a KeyError would raise (:issue:`16748`)
+- Bug in :meth:`DataFrame.plot` was overwriting matplotlib's shared y axes behaviour when no sharey parameter was passed (:issue:`37942`)
```
[the remark this answers is not in the document]
Comment (lrusnac) [reply doc/source/whatsnew/v1.2.0.rst:667]: yeah, they both share the same method in the end

**Comment by lrusnac** [comment]
> Thanks for the PR!
> 
> I just have one minor comment on whatsnew, plus a general question: what if users explicitly specify `sharey=False` in `df.plot`? Will `sharey` previously defined by matplotlib be overridden by pandas (I know such a setting is very silly)?
> 
> something like below using example from your test:
> 
> ```python
> df = DataFrame(np.random.rand(4, 2), columns=["x", "y"])
> fig, (ax1, ax2) = plt.subplots(1, 2, sharey=True)
> 
> df.plot(ax=ax1, kind="area", sharey=False)
> df.plot(ax=ax2, kind="area", sharey=False)
> ```

that's a fair question, to me that doesn't make much sense, I see the use of sharey in the plot only when it's pandas to make multiple plots, if you specify an ax then it's a bit weird option in my interpretation. 

that said, the line plots for instance are not overwriting the behaviour, it's specific to area plots where in the end it sets the min or max limit to 0, and in the process resets the other extreme too.

that's all to say that if you pass a specific False, then pandas should maybe overwrite the default with False (to be tested how the line plots handle that actually).

**Comment by lrusnac** [comment]
I checked, apparently `sharey=False` is ignored for other plot types
Screenshot 2020-11-19 at 20 55 04

**Comment by lrusnac** [comment]
to be honest I don't know why this snippet 

```python
if self.ylim is None:
    if (data >= 0).all().all():
        ax.set_ylim(0, None)
    elif (data <= 0).all().all():
```

would be needed at all. if it's area chart, it does use line plots for 0 and the other extreme and the fill between, so it should be already set as min or max limit at 0. but maybe I don't have a good understanding of the system

**Comment by charlesdong1991** [comment]
> to me that doesn't make much sense, I see the use of sharey in the plot only when it's pandas to make multiple plots

As I said, I also don't think this makes much sense, and totally agree with you on the assessment since `sharex or y` should only apply when `subplots=true`. I was just curious what the result would be if doing so. 

> would be needed at all. if it's area chart, it does use line plots for 0 and the other extreme and the fill between, so it should be already set as min or max limit at 0.

unfortunately, I am not very familiar with this snippet either, cannot answer you right now, need a look. but if you could find a way of cleaning this up, feel free to come up with a new PR to do so!

**Comment by charlesdong1991** [comment]
very nice @lrusnac and any clean-up PR is very welcome!

**Pull request merged**